relevanceai/dataset/write/write.py

```diff
@@ -952,3 +1008,35 @@ def insert_list(self, labels: list, label_field: str = "label", **kwargs):
         """
         documents = [{label_field: l} for l in labels]
         return self.insert_documents(documents=documents, **kwargs)
+
+    def batched_upsert_media(
+        self,
+        images: List[str],
+        show_progress_bar: bool = False,
+    ) -> List[str]:
+        """batched upsert media"""
+
+        bs = int(len(images) / (os.cpu_count() + 4))
```

Comment:
Assuming `bs` is `batch_size` why is it `cpu_count` + 4 ?

Reply:
Will need to check with the API team on rate limiting this

Reply:
in `ThreadPoolExectuor`, if `max_workers` is `None`, then it will spawn `os.cpu_count() + 4` workers. I set batch size as to evenly distribute the number of images to upload across all workers.